src/MultiSelect/MultiSelect.tsx

```diff
@@ -220,33 +233,30 @@ const MultiSelect: React.FC<Props> = ({
         {...getMenuProps()}
       >
         {isOpen &&
-          (selectableOptions.length > 0 ? (
-            selectableOptions.map((o, i) => (
-              <li
-                className={classNames(
-                  cssClass.MENU_OPTION,
-                  i === highlightedIndex && cssClass.MENU_OPTION_HIGHLIGHTED,
-                )}
-                key={`${o.value}${i}`}
-                {...getItemProps({ item: o, index: i })}
-              >
-                {o.content || o.value}
-              </li>
-            ))
+          // slice out the dummy "add item X" option
+          (selectableOptions.slice(1).length === 0 && (!creatable || hasExactMatch) ? (
+            <li className={classNames(cssClass.NO_OPTIONS_FOUND)}>No options</li>
           ) : (
-            <div
-              className={classNames(!creatable && cssClass.NO_OPTIONS_FOUND)}
-              onClick={() => {
-                if (creatable) {
-                  const newOption: Option = { value: inputValue };
-                  setOptions(options.concat(newOption));
-                  setInputValue("");
-                  addSelectedItem(newOption);
-                }
-              }}
-            >
-              {creatable ? `Create new item ${inputValue}` : "No options"}
-            </div>
+            selectableOptions.map((o, i) => {
+              const isAddNewItemOption = o.value === ADD_NEW_ITEM_KEY;
+              // hide the first dummy "add item X" option if not eligible
+              if (isAddNewItemOption && (inputValue.length === 0 || !creatable || hasExactMatch)) {
```

Comment:
```suggestion
              if (isAddNewItemOption && (inputValue === "" || !creatable || hasExactMatch)) {
```
This way would be more idiomatic to check if `inputValue` is empty

Reply:
Hmm yeah maybe I can do `!inputValue`